src/main/java/com/game/lotto/prize/Rank.java

```diff
@@ -24,11 +24,12 @@ public long getMoneyAmount() {
         return moneyAmount;
     }
 
-    public static Rank valueOf(int strikes) {
-        if(strikes == 6) return FIRST_PRIZE_MONEY_AMOUNT_WITH_6_STRIKES;
-        if(strikes == 5) return THIRD_PRIZE_MONEY_AMOUNT_WITH_5_STRIKES;
-        if(strikes == 4) return FOURTH_PRIZE_MONEY_AMOUNT_WITH_4_STRIKES;
-        if(strikes == 3) return FIFTH_PRIZE_MONEY_AMOUNT_WITH_3_STRIKES;
+    public static Rank valueOf(int strikes, boolean containsBonusBall) {
+        if (strikes == 6) return FIRST_PRIZE_MONEY_AMOUNT_WITH_6_STRIKES;
+        if (strikes == 5 && containsBonusBall) return SECOND_PRIZE_MONEY_AMOUNT_WITH_5_STRIKES_AND_BONUS;
+        if (strikes == 5) return THIRD_PRIZE_MONEY_AMOUNT_WITH_5_STRIKES;
+        if (strikes == 4) return FOURTH_PRIZE_MONEY_AMOUNT_WITH_4_STRIKES;
+        if (strikes == 3) return FIFTH_PRIZE_MONEY_AMOUNT_WITH_3_STRIKES;
         return NONE;
     }
```

Comment:
만약 해당 enum의 객체들이 bonusBall을 가져야 하는 여부를 필드로 가진다면 지금처럼 조건문을 나열하지 않고도 해결이 가능할 것 같아요!!

```suggestion
    // 보너스볼 포함 필요 여부를 needBonusBall 필드로 관리한다고 가정
    
    public static Rank valueOf(int strikes, boolean containsBonusBall) {
        if (strikes == 5 && containsBonusBall) {
            return SECOND_PRIZE_MONEY_AMOUNT_WITH_5_STRIKES_AND_BONUS;
        }
        
        return Arrays.stream(values())
                .filter(rank -> !rank.needBonusBall) // 보너스볼 불 필요
                .filter(rank -> rank.strikes == strikes)
                .findAny()
                .orElse(NONE);
    }
```